src/app/frontend/replicationcontrollerdetail/replicationcontrollerinfo_controller.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  stateName as replicationcontrollers,
+} from 'replicationcontrollerlist/replicationcontrollerlist_state';
+
+/**
+ * Controller for the replication controller info directive.
+ * @final
+ */
+export default class ReplicationControllerInfoController {
+  /**
+   *
+   * @param {!./replicationcontrollerdetail_state.StateParams} $stateParams
+   * @param {!ui.router.$state} $state
+   * @param {!angular.$log} $log
+   * @param {!./replicationcontroller_service.ReplicationControllerService}
+   * kdReplicationControllerService
+   *
+   * @ngInject
+   */
+  constructor($stateParams, $state, $log, kdReplicationControllerService) {
+    /** @private {!./replicationcontrollerdetail_state.StateParams} */
+    this.stateParams_ = $stateParams;
+
+    /** @private {!./replicationcontroller_service.ReplicationControllerService} */
+    this.kdReplicationControllerService_ = kdReplicationControllerService;
+
+    /** @private {!ui.router.$state} */
+    this.state_ = $state;
+
+    /** @private {!angular.$log} */
+    this.log_ = $log;
+
+    /**
+     * Replication controller details, contains all the information about it (two-way binding).
```

Comment:
Is this two way? I'd say this is "Initialized from the scope" and that's all. You don publish any changes back, don't you?